test/unittests/test_intx_api.cpp

```diff
@@ -259,6 +259,54 @@ TYPED_TEST(uint_api, comparison)
     EXPECT_TRUE(d >= x);
 }
 
+struct W
+{
+    intx::uint256 v;
+
+    friend auto operator<=>(W, W) = default;
+};
+
+struct W128
+{
+    intx::uint128 v;
+    friend auto operator<=>(W128, W128) = default;
+};
+
+TYPED_TEST(uint_api, spaceship_operator)
+{
+    EXPECT_EQ(1, 1);
+
+    auto a = W{uint256{1}};
```

Comment:
There are typed tests and the type is named `TypeParam`. So you can do:

```cpp
Wrapper a{TypeParam{1}};

Reply:
if I understood correctly, there is no need to repeat the test cases for uint128 and other types. the test framework will iterate through all of them for all test cases.  is it correct?

Reply:
Correct.